tests/test_packages/test_connections/test_p2p_libp2p/test_slow_queue.py

```diff
@@ -0,0 +1,158 @@
+            cls.bad_address = _make_libp2p_connection(
+                data_dir=temp_dir_gen, port=port_genesis
+            ).node.address
+            cls.connection_genesis = _make_libp2p_connection(
+                data_dir=temp_dir_gen, port=port_genesis
+            )
+            cls.multiplexer_genesis = Multiplexer(
+                [cls.connection_genesis], protocols=[MockDefaultMessageProtocol]
+            )
+            cls.log_files.append(cls.connection_genesis.node.log_file)
+            cls.multiplexer_genesis.connect()
+            cls.multiplexers.append(cls.multiplexer_genesis)
+
+            genesis_peer = cls.connection_genesis.node.multiaddrs[0]
+
+            cls.connections = [cls.connection_genesis]
+
+            temp_dir = os.path.join(cls.t, "temp_dir_100")
+            os.mkdir(temp_dir)
+
+            cls.conn = _make_libp2p_connection(
+                data_dir=temp_dir, port=port_genesis + 100, entry_peers=[genesis_peer]
+            )
+
+            port = port_genesis
+            for i in range(2):
+                port += 1
+                temp_dir = os.path.join(cls.t, f"temp_dir_{i}")
+                os.mkdir(temp_dir)
+                conn = _make_libp2p_connection(
+                    data_dir=temp_dir, port=port, entry_peers=[genesis_peer]
+                )
+                mux = Multiplexer([conn], protocols=[MockDefaultMessageProtocol])
+
+                cls.connections.append(conn)
+
+                cls.log_files.append(conn.node.log_file)
+                mux.connect()
+                cls.multiplexers.append(mux)
+
+            for conn in cls.connections:
+                assert conn.is_connected is True
+        except Exception as e:
+            cls.teardown_class()
+            raise e
+
+    @pytest.mark.asyncio
+    async def test_slow_queue(self):
+        """Test slow queue."""
+        con2 = self.connections[-1]
+        await self.conn.connect()
+
+        def _make_envelope(addr):
+            msg = DefaultMessage(
+                dialogue_reference=("", ""),
+                message_id=1,
+                target=0,
+                performative=DefaultMessage.Performative.BYTES,
+                content=b"hello",
+            )
+
+            envelope = Envelope(to=addr, sender=self.conn.node.address, message=msg,)
+            return envelope
+
+        try:
+            for _ in range(50):
+                for addr in [con2.node.address, self.bad_address]:
+                    await self.conn._node_client.send_envelope(_make_envelope(addr))
+
+            for _ in range(2):
+                for addr in [self.bad_address, con2.node.address]:
+                    await self.conn._node_client.send_envelope(_make_envelope(addr))
+
+            def _check():
+                with open(self.conn.node.log_file) as f:
+                    return "while sending slow envelope:" in f.read()
+
+            wait_for_condition(_check, timeout=30, period=1)
```

Comment:
As I understand it, this will throw a timeout exception if the slow queue is not used, but since this is inside a try-except, won't the test always pass?

Reply:
it's try..finally to disconnect the connection in case of fail, no exceptions handled or supressed